server/graphql/v2/object/Host.js

```diff
@@ -67,6 +67,16 @@ const getNumberOfDays = (startDate, endDate, host) => {
   return days(since, endDate || undefined);
 };
 
+const getTimeUnit = numberOfDays => {
+  if (numberOfDays <= 7) {
+    return 'WEEK';
+  } else if (numberOfDays <= 366) {
+    return 'MONTH';
+  } else {
+    return 'YEAR';
+  }
```

Comment:
```suggestion
  if (numberOfDays <= 21) {
    return 'DAY'; // Up to 3 weeks
  } else if (numberOfDays <= 90) {
    return 'WEEK'; // Up to 3 months
  } else if (numberOfDays <= 365 * 3) {
    return 'MONTH'; // Up to 3 years
  } else {
    return 'YEAR';
  }
```

Reply:
@Betree : The requirement to show the weekly amounts seems a bit complicated to wrap my head around. For now I've added,

https://github.com/opencollective/opencollective-api/blob/34c1e2a719caa1f7f56241af27cd674aeaee2390/server/graphql/v2/object/Host.js#L71-L77

I think that's better that what was there hopefully. If no objections, I'll do this weekly amount display in a separate PR? 🤔

Reply:
If you do `numberOfDays <= 31` you'll end up with the same issue of having a chart that only contains a single value. Play with it to find something that looks nice, but something like `90` sounds like a decent default.

No problem to add `WEEK` in a separate PR if it complicates things (and actually, we should only add it if there's a value in doing so). Could you shortly explain where the complexity to add this is coming from?

Reply:
Actually looking at this today I realized the front-end implementation for this dynamic timeunits can be greatly simplified. I've just did that and now it should look much better and also I've added the week support. 🙂 Sorry about this, I feel I was making a too complicated mess out of this. 🤔